x-pack/plugins/cases/server/routes/api/files/post_file.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import Boom from '@hapi/boom';
+import mime from 'mime-types';
+import { parse } from 'path';
+
+import { schema } from '@kbn/config-schema';
+import { ReplaySubject } from 'rxjs';
+
+import { AbortedUploadError } from '@kbn/files-plugin/server/file/errors';
+import type { HapiReadableStream } from '../../../client/attachments/types';
+import type { attachmentApiV1 } from '../../../../common/types/api';
+
+import { CASE_FILES_URL, MAX_FILE_SIZE } from '../../../../common/constants';
+import { createCaseError } from '../../../common/error';
+import { createCasesRoute } from '../create_cases_route';
+import type { caseDomainV1 } from '../../../../common/types/domain';
+
+export const postFileRoute = createCasesRoute({
+  method: 'post',
+  path: CASE_FILES_URL,
+  params: {
+    params: schema.object({
+      case_id: schema.string(),
+    }),
+  },
+  routerOptions: {
+    access: 'public',
+    summary: 'Attach a file to a case',
+    tags: ['oas-tag:cases'],
+    body: {
+      /*
+       * The maximum file size allowed here is the upper limit allowed when registering the cases file
+       * kinds. We register the file kinds in x-pack/plugins/cases/public/files/index.ts.
+       *
+       * If the user configured a custom value the validation will still fail when calling the service.
+       */
+      maxBytes: MAX_FILE_SIZE,
```

Comment:
Users can configure the max size of files through `kibana.yml`. Also, the max size of an image is different from the size of the other files. Maybe validation is not needed on the route level because it will performed by the File Service. We define our validation here `x-pack/plugins/cases/server/files/index.ts`. Integration tests verifying that should be enough.